### PR TITLE
Check aggregation length against the wider cidr generated from range

### PR DESCRIPTION
**Pull request by fedepaol**
If we specify addresses in a pool by range (and not by cidr), the check
that the lenght of the cidr is bigger than the aggregation lenght is
performed against all the possible cidrs that range translates to, and
it is misleading. The check makes sense when we declare a cidr, because
we know we won that whole cidr, but in the other case we may be using a
subset of a cidr for other reasons, and the check here leads to false
negatives.

A scenario like the one in the test that is being added should not fail:
3.3.3.2-3.3.3.254 with aggregation length 26

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -424,21 +424,26 @@ func parseAddressPool(p addressPool, bgpCommunities map[string]uint32) (*Pool, e
 	if len(p.Addresses) == 0 {
 		return nil, errors.New("pool has no prefixes defined")
 	}
+
```
Comment (russellb) [inline internal/config/config.go:427]: s/won/own/ in the commit msg

**Review by russellb — COMMENTED** [review]
small typo. otherwise lgtm

**Comment by msherif1234** [comment]
its very odd that we have check for one addresspool fmt but not the other, look like a kludge to me with either remove the check or find a better way to do it

File: internal/config/config.go [hunk 2]
```diff
@@ -525,6 +530,7 @@ func parseBGPAdvertisements(ads []bgpAdvertisement, cidrs []*net.IPNet, communit
 			if cidr.IP.To4() == nil {
 				maxLength = ad.AggregationLengthV6
 			}
+
```
[the remark this answers is not in the document]
Comment (fedepaol) [reply internal/config/config.go:533]: Because they are added all of them to the same slice

**Comment by fedepaol** [comment]
> its very odd that we have check for one addresspool fmt but not the other, look like a kludge to me, we either remove the check or find a better way to do it

Yep, but I think adding the check when specifying a cidr has some value. Another thing we can do is, when we have multiple cidrs to filter against the largest one. It will be more correct, probably as much as clunky.

File: internal/config/config.go (hunk 1 above)
Comment (oribon) [reply internal/config/config.go:427]: Also lenght -> length

Comment (fedepaol) [reply internal/config/config.go:427]: Fixed (it's one of those things I never get right)

File: internal/config/config.go [hunk 3]
```diff
@@ -629,6 +637,20 @@ func cidrContainsCIDR(outer, inner *net.IPNet) bool {
...
 
+func lowestMask(cidrs []*net.IPNet) int {
+	if len(cidrs) == 0 {
+		return 0
+	}
+	lowest, _ := cidrs[0].Mask.Size()
+	for _, c := range cidrs {
+		s, _ := c.Mask.Size()
+		if lowest > s {
+			lowest, _ = c.Mask.Size()
```
Comment (oribon) [inline internal/config/config.go:648]: Why not `lowest = s`?

Comment (fedepaol) [reply internal/config/config.go:648]: `internal/config/config.go:648:11: assignment mismatch: 1 variable but c.Mask.Size returns 2 values`

Comment (fedepaol) [reply internal/config/config.go:648]: Sorry, now I get it. You are right.

Comment (fedepaol) [reply internal/config/config.go]: Fixed

**Comment by oribon** [comment]
lgtm

**Pull request merged**